src/System.Transactions.Local/src/System/Transactions/TransactionState.cs

```diff
@@ -1683,7 +1666,7 @@ internal override void CheckForFinishedTransaction(InternalTransaction tx)
 
         internal override void GetObjectData(InternalTransaction tx, SerializationInfo serializationInfo, StreamingContext context)
         {
-            throw TransactionInDoubtException.Create(TraceSourceType.TraceSourceBase, SR.TransactionIndoubt, tx._innerException, tx.DistributedTxId);
+            throw new PlatformNotSupportedException();
```

Comment:
Same comment about preserving previous code in a comment. It really applies to all the code that you have replaced with a PlatformNotSupportedException in System.Transactions.